mtt-management/src/mtt_types.rs

```diff
@@ -314,8 +314,133 @@ pub fn set_4k_page_type(
 fn get_mtt_base() -> u64 {
     todo!("Implement code to get MTT base address");
 }
+
+// Testing only function intended to emulate functionality
+// to get the MTT base.
+#[cfg(not(all(target_arch = "riscv64", target_os = "none")))]
+fn get_mtt_base() -> u64 {
+    static mut MTT_BASE: [u8; 1024 * 1024 * 8] = [0; 1024 * 1024 * 8];
+    // Not safe: Intended for testing purposes only
+    let addr = unsafe { &MTT_BASE as *const u8 };
+    addr as u64
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use riscv_pages::*;
+    #[test]
+    fn test_1gb_ranges() {
+        for i in 0u64..1024 {
+            let addr = 0u64 + (i * 1024 * 1024 * 1024);
+            let entry_type = get_l2_entry(addr);
+            assert!(matches!(entry_type, MttL2EntryType::MttNonConfidential1G));
+            let supervisor_page = PageAddr::new(SupervisorPhysAddr::supervisor(addr)).unwrap();
+            set_1gb_range_type(supervisor_page, MttMemoryType::Confidential);
+            let entry_type = get_l2_entry(addr);
+            assert!(matches!(entry_type, MttL2EntryType::MttConfidential1G));
+            set_1gb_range_type(supervisor_page, MttMemoryType::NonConfidential);
+            let entry_type = get_l2_entry(addr);
+            assert!(matches!(entry_type, MttL2EntryType::MttNonConfidential1G));
+        }
+        test_64mb_ranges();
+        test_4k_ranges();
+    }
+
+    fn test_64mb_ranges() {
+        let entry_type = get_l2_entry(0);
+        assert!(matches!(entry_type, MttL2EntryType::MttNonConfidential1G));
+        let supervisor_page = PageAddr::new(SupervisorPhysAddr::supervisor(0)).unwrap();
+        fracture_1gb_range(supervisor_page);
+        for i in 0u64..=15 {
+            let addr = 0u64 + (i * 1024 * 1024 * 64);
+            let entry_type = get_l2_entry(addr);
+            assert!(matches!(entry_type, MttL2EntryType::Mtt2MBPages(0)));
+        }
+        let supervisor_page = PageAddr::new(SupervisorPhysAddr::supervisor(0)).unwrap();
+        set_1gb_range_type(supervisor_page, MttMemoryType::Confidential);
+        let entry_type = get_l2_entry(0);
+        assert!(matches!(entry_type, MttL2EntryType::MttConfidential1G));
+        fracture_1gb_range(supervisor_page);
+        for i in 0u64..=15 {
+            let addr = 0u64 + (i * 1024 * 1024 * 64);
+            let entry_type = get_l2_entry(addr);
+            assert!(matches!(
+                entry_type,
+                MttL2EntryType::Mtt2MBPages(0xffff_ffff)
+            ));
+        }
+        set_1gb_range_type(supervisor_page, MttMemoryType::NonConfidential);
+        let entry_type = get_l2_entry(0);
+        assert!(matches!(entry_type, MttL2EntryType::MttNonConfidential1G));
+    }
+
+    fn test_4k_ranges() {
+        static mut L1_PAGE: [u8; 4096 * 2] = [0; 4096 * 2];
+        // Not safe: Intended for testing purposes only
+        let addr = unsafe { &L1_PAGE as *const u8 } as u64;
+        let l1_page = riscv_pages::PageSize::Size4k.round_up(addr as u64);
```

Comment:
Alternatively might use std::alloc::alloc to get an aligned allocation? Or std::slice::align_to if you want to stick with the static allocation? Or #[repr(align(4096))], but that requires a wrapper type IIUC.

Reply:
Sure...it was intended only for testing purposes (and we normally don't use std:: functions).